### PR TITLE
Add Black, flake8, isort and pre-commit

### LINKED ISSUE #6: Add lint etc. workflows and hooks for cleaning the code
- Black
- Pylint
- Pep8

TBA

### PR DESCRIPTION
**Pull request by Joonalai**
Utilize pre-commit via Github Actions and add instructions for developers to use those tools locally.

Resolves #6.

### CODE REVIEW

File: infrastructure/template/plugin/.editorconfig [hunk 1]
```diff
@@ -9,6 +9,9 @@ insert_final_newline = true
 [*.py]
 indent_style = space
 indent_size = 4
+# Black compatible values for isort https://black.readthedocs.io/en/stable/compatible_configs.html#isort
+profile = black
```
Comment (LKajan) [inline infrastructure/template/plugin/.editorconfig:13]: These should probably be in setup.cfg [isort] block?

File: infrastructure/template/root/setup.cfg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[flake8]
```
Comment (LKajan) [inline infrastructure/template/root/setup.cfg:1]: Should the [isort] config be added here?

File: infrastructure/template/root/setup.cfg [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+# Black compatible values https://black.readthedocs.io/en/stable/compatible_configs.html#flake8
+max-line-length = 88
+extend-ignore =
+            E203, # whitespace before ':'
+            W503  # line break before binary operator
```
Comment (LKajan) [inline infrastructure/template/root/setup.cfg:6]: W503 is ignored by default by flake8. So no need to specifically ignore.
https://github.com/psf/black/pull/1661

**Review by LKajan — DISMISSED** [review]
isort configuration is now in .editorconfig. Otherwise looks goot to me.

File: infrastructure/template/plugin/.editorconfig (hunk 1 above)
Comment (Joonalai) [reply infrastructure/template/plugin/.editorconfig:13]: Unfortunately isort ignores setup.cfg if .editorconfig is present.

File: infrastructure/template/root/setup.cfg (hunk 3 above)
Comment (Joonalai) [reply infrastructure/template/root/setup.cfg:6]: Good catch! I will remove that.

File: infrastructure/template/plugin/.editorconfig (hunk 1 above)
Comment (LKajan) [reply infrastructure/template/plugin/.editorconfig:13]: Could you verify that again? According to https://pycqa.github.io/isort/docs/configuration/config_files/ and to my experience setup.cfg supersedes .editorconfig which is the last hope for isort.

Comment (Joonalai) [reply infrastructure/template/plugin/.editorconfig]: Yeah, I realized that this was caused by .editorconfig lying in the deeper lever in the file structure. I moved it next to the setup.cfg and it worked right away! :)

**Pull request merged**